auth/src/main/java/com/firebase/ui/auth/KickoffActivity.java

```diff
@@ -41,6 +42,9 @@ protected void onSuccess(@NonNull IdpResponse response) {
             protected void onFailure(@NonNull Exception e) {
                 if (e instanceof UserCancellationException) {
                     finish(RESULT_CANCELED, null);
+                } else if (e instanceof FirebaseAuthAnonymousUpgradeException) {
+                    IdpResponse res = ((FirebaseAuthAnonymousUpgradeException) e).getResponse();
+                    finish(RESULT_CANCELED, new Intent().putExtra(ExtraConstants.IDP_RESPONSE, res));
```

Comment:
This is wrong 😊. The `FirebaseAuthAnonymousUpgradeException` should extend `FirebaseUiException` so all of this is handled for you correctly. You should be able to remove this whole block of code.

Reply:
Not sure how that would help. Can you explain? :smile:

Reply:
Oh hold up, I think I'm wrong actually. I thought you were repackaging the exception, but you're actually extracting the response. Nevermind then. 😊

@samtstern @lsirac side note: this is another argument in favor of https://github.com/firebase/FirebaseUI-Android/pull/1358#discussion_r195985936. If we went with that idea, we should be able to just return the exception in a standard error response and call it good.